addons/web_editor/controllers/main.py

```diff
@@ -537,7 +537,17 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
             fields['res_id'] = res_id
         if name:
             fields['name'] = name
-        attachment = attachment.copy(fields)
+
+        # Restricted editors can handle attachments related to records to which they have access
+        # Would user be able to read fields of original record ?
+        if attachment and attachment.res_model and attachment.res_id:
+            request.env[attachment.res_model].browse(attachment.res_id).with_user(request.env.user).check_access_rights('read')
+
+        # Would user be able to write fields of target record ?
+        # Rights check works with res_id=0 because browse(0) returns an empty record set
+        request.env[fields['res_model']].browse(fields['res_id']).with_user(request.env.user).check_access_rights('write')
```

Comment:
Why is the `with_user()` needed here? The route / rpc can't be done / call with another user than yourself AFAIK?